freqtrade/leverage/interest.py

```diff
@@ -0,0 +1,43 @@
+from decimal import Decimal
+from math import ceil
+
+from freqtrade.exceptions import OperationalException
+
+
+one = Decimal(1.0)
+four = Decimal(4.0)
+twenty_four = Decimal(24.0)
+
+
+def interest(
+    exchange_name: str,
+    borrowed: Decimal,
+    rate: Decimal,
+    hours: Decimal
+) -> Decimal:
+    """Equation to calculate interest on margin trades
+
+
+    :param exchange_name: The exchanged being trading on
+    :param borrowed: The amount of currency being borrowed
+    :param rate: The rate of interest
+    :param hours: The time in hours that the currency has been borrowed for
+
+    Raises:
+        OperationalException: Raised if freqtrade does
+        not support margin trading for this exchange
+
+    Returns: The amount of interest owed (currency matches borrowed)
+
+    """
+    exchange_name = exchange_name.lower()
+    if exchange_name == "binance":
+        return borrowed * rate * ceil(hours)/twenty_four
+    elif exchange_name == "kraken":
+        # Rounded based on https://kraken-fees-calculator.github.io/
```

Comment:
If the calculations are expected to be completely different ...
maybe it would make sense to move this to the respective exchange-classes? 

The only premise for that would be that we should try not to call that from persistence (which could be a dealbreaker, and which is why i'm posting it hree as a question).

Reply:
Excluding it from persistence is the only reason It's not just in the exchange class